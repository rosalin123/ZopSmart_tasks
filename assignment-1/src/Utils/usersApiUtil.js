const baseUrl = 'https://jsonplaceholder.typicode.com';

export const fetchUsers = async () => {
  try {
    let url = `${baseUrl}/users`;
    let users = await fetch(url, { method: 'GET' });
    let usersJson = await users.json();
    return usersJson;
  } catch (err) {
    console.log('Error fetching users details', err.stack);
  }
};

export const getUser = async (id) => {

    let url = `${baseUrl}/users/${id}`;
    let user = await fetch(url, { method: 'GET' });
    let userJson = await user.json();
    return userJson;

};
