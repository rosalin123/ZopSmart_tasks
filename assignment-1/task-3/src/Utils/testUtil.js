import React from 'react';

export const findbyTestAttr = (component, attr) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
};
