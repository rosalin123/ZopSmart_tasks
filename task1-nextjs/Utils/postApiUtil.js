import axios from 'axios';
import fetch from 'isomorphic-unfetch';
const baseUrl = 'https://jsonplaceholder.typicode.com';

export const fetchPosts = async () => {
  let url = `${baseUrl}/posts`;
  let posts = await axios.get(url);
  let postsJson = await posts.json();
  return postsJson;
};

export const getPost = async (id) => {
  let url = `${baseUrl}/posts/${id}`;
  let post = fetch(url);
  let postJson = await post.json();
  return postJson;
};

export const getPostComments = async (id) => {
  let url = `${baseUrl}/posts/${id}/comments`;
  let comments = await axios.get(url);
  let commentsJson = await comments.json();
  return commentsJson;
};

export const getUserPosts = async (id) => {
  let url = `${baseUrl}/posts?userId=${id}`;
  let userPosts = await fetch(url);
  let userPostsJson = await userPosts.json();
  return userPostsJson;
};
