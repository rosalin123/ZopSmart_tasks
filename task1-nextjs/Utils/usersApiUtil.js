import fetch from 'isomorphic-unfetch';
const baseUrl = 'https://jsonplaceholder.typicode.com';

export const fetchUsers = async () => {
  try {
    let url = `${baseUrl}/users`;
    let users = await fetch(url);
    let usersJson = await users.json();
    return usersJson;
  } catch (err) {
    console.log('Error fetching users details', err.stack);
  }
};

export const getUser = async (id) => {
  let url = `${baseUrl}/users/${id}`;
  let user = await fetch(url);
  let userJson = await user.json();
  return userJson;
};
