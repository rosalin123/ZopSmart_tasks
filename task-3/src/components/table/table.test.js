import React from 'react';
import { shallow, mount, render } from 'enzyme';
import { Table, mapDispatchToProps } from './index';
import { findbyTestAttr } from '../../Utils/testUtil';

const setUp = (props) => {
  const component = shallow(<Table {...props} />);
  return component;
};

describe('TableContainer component', () => {
  let component;
  let props = {
    rows: 2,
    columns: 2,
    values: [
      [0, 0],
      [0, 0],
    ],
    sumProduct: [
      [0, 0],
      [0, 0],
    ],
    updateValues: jest.fn(),
  };

  beforeEach(() => {
    component = setUp(props);
  });

  it('Table should render without throwing any error', () => {
    expect(component).toMatchSnapshot();
  });

  it('mapDispatchToProps', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).updateValues(props.values, props.sumProduct);
    expect(dispatch.mock.calls[0][0]).toEqual({
      type: 'UPDATE_VALUES',
      payload: { values: props.values, sumProduct: props.sumProduct },
    });
  });

  it('componentDidMount must call methods as expected', () => {
    const instance = component.instance();

    jest.spyOn(instance.props, 'updateValues');
    instance.componentDidMount();
    expect(instance.props.updateValues).toBeCalledTimes(1);
  });

  it('updateValues method must be called when an onchange event is triggered', () => {
    const instance = component.instance();
    const event = { target: { dataset: { row: 1, column: 1 } }, value: 2 };
    const { row, column } = event.target.dataset;
    let { values, sumProduct } = props;
    let prevValue = values[row][column];
    values[row][column] = Number(event.target.value);
    sumProduct[row][0] =
      sumProduct[row][0] - prevValue + Number(event.target.value);
    sumProduct[row][1] = values[row].reduce((product, value) => {
      product *= value;
      return product;
    }, 1);
    instance.handleChange(event);
    expect(instance.props.updateValues).toBeCalledWith(values, sumProduct);
  });

  it('should handle change on input', () => {
    const instance = component.instance();
    jest.spyOn(instance.props, 'updateValues');
    jest.spyOn(instance, 'handleChange');
    const wrapper = findbyTestAttr(component, 'input');
    const event = { target: { dataset: { row: 1, column: 1 } }, value: 2 };
    wrapper.at(0).simulate('change', event);
    wrapper.at(1);
    expect(instance.handleChange).toBeCalledTimes(1);
    expect(instance.props.updateValues).toBeCalled();
  });
});
