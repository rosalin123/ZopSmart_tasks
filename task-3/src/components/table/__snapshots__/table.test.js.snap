// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableContainer component Table should render without throwing any error 1`] = `
<div>
   
  <table
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <tbody>
      <tr>
        <th>
          S.No
        </th>
        <th
          key="0"
        >
          Val 
          1
        </th>
        <th
          key="1"
        >
          Val 
          2
        </th>
        <th
          key="100"
        >
          Sum
        </th>
        <th
          key="101"
        >
          Product
        </th>
      </tr>
      <tr
        data-row="index"
        key="0"
      >
        <td>
          1
        </td>
        <td
          key="0"
        >
          <input
            data-column={0}
            data-row={0}
            data-test="input"
            defaultValue={0}
            onChange={[Function]}
            type="number"
          />
        </td>
        <td
          key="1"
        >
          <input
            data-column={1}
            data-row={0}
            data-test="input"
            defaultValue={0}
            onChange={[Function]}
            type="number"
          />
        </td>
        <td
          key="100"
        >
          0
        </td>
        <td
          key="111"
        >
          0
        </td>
      </tr>
      <tr
        data-row="index"
        key="1"
      >
        <td>
          2
        </td>
        <td
          key="0"
        >
          <input
            data-column={0}
            data-row={1}
            data-test="input"
            defaultValue={0}
            onChange={[Function]}
            type="number"
          />
        </td>
        <td
          key="1"
        >
          <input
            data-column={1}
            data-row={1}
            data-test="input"
            defaultValue={0}
            onChange={[Function]}
            type="number"
          />
        </td>
        <td
          key="100"
        >
          0
        </td>
        <td
          key="111"
        >
          0
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
